test/safe-mode-check.js

```diff
@@ -1,33 +1,32 @@
+#!/usr/bin/env node
+
 import {contextsDir, credentialsDir} from '../lib/index.js';
 import EleventyFetch from '@11ty/eleventy-fetch';
 import fs from 'node:fs';
 import {glob} from 'glob';
 import jsonld from 'jsonld';
 import {JsonLdDocumentLoader} from 'jsonld-document-loader';
 
+// Pass in `extract` to get context list instead of safe mode check results
+const method = process.argv.at(2);
+
 const jdl = new JsonLdDocumentLoader();
 
-// Loop all local context files and add them as static contexts at the base URL
-const paths = (await glob([`${contextsDir}/**/*.json`],
-  {stat: true, withFileTypes: true}))
-  .map(path => path.fullpath());
-
-const pathParts = paths.map(path => {
-  const parts = path.split('/');
-  return [parts[parts.length - 2], parts[parts.length - 1]];
-});
-
-pathParts.forEach(([dir, file]) => {
-  const context = JSON.parse(
-    fs.readFileSync(`${contextsDir}/${dir}/${file}`));
-  const oldUrl = `https://contexts.vcplayground.org/examples/${dir}/${file}`;
-  jdl.addStatic(oldUrl, context);
-  const newUrl = `https://examples.vcplayground.org/contexts/${dir}/${file}`;
-  jdl.addStatic(newUrl, context);
-});
+// collection to track all loaded contexts
+const contextUrls = new Set();
+
 jdl.setProtocolHandler({protocol: 'https',
   handler: {
     async get({url}) {
+      if(url.startsWith('https://examples.vcplayground.org/contexts/')) {
+        const regex =
+          /https\:\/\/examples\.vcplayground\.org\/contexts\/(.*)\/(.*)$/;
+        const [, dir, file] = url.match(regex);
+        const context = JSON.parse(
+          fs.readFileSync(`${contextsDir}/${dir}/${file}`));
+        jdl.addStatic(url, context);
+      }
```

Comment:
This seems a little weird ... like it will load it from disk on first request, but then fetch from the Web anyway and potentially return something different (if there's an unlikely mismatch). I'd expect it to just return what was loaded.

Not blocking over this though, +1 to get this merged. If this changes to return what we loaded it's unnecessary to ask for another re-review from me.

Reply:
Yeah...that's a bug. I'll fix it soon...but not in this PR.